rotkehlchen/chain/ethereum/balancer/typing.py

```diff
@@ -0,0 +1,74 @@
+from typing import (
+    Any,
+    DefaultDict,
+    Dict,
+    List,
+    NamedTuple, NewType,
+    Optional,
+    Set,
+)
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.fval import FVal
+from rotkehlchen.typing import ChecksumEthAddress
+
+
+# Below set of classes where to store PoolShare, Pool and PoolToken data
+# BalancerPoolAsset stores PoolToken data, and adds custom ones
+class BalancerPoolAsset(NamedTuple):
+    address: ChecksumEthAddress  # PoolToken.id, token contract address
+    balance: FVal  # PoolToken.balance
+    denorm_weight: FVal  # PoolToken.denormWeight
+    name: str  # PoolToken.name
+    symbol: str  # PoolToken.symbol
+    # Custom fields
+    user_balance: FVal  # Estimated token balance
+    user_balance_usd: Optional[FVal] = ZERO  # Estimated token balance in USD
+    asset: Optional[Asset] = None
+    asset_usd: Optional[FVal] = ZERO  # price in USD per token
+
+
+# BalancerPool stores PoolShare and Pool data
+# ! TODO VN PR: currently `asset` field can't be populated with <Asset> because
+# ! there are as many BPT contracts as pools in Balancer
+class BalancerPool(NamedTuple):
+    address: ChecksumEthAddress  # Pool.id, pool contract address
+    assets: List[BalancerPoolAsset]  # Pool.tokens
+    assets_count: FVal  # Pool.tokensCount
+    symbol: str  # Pool.symbol
+    balance: FVal  # Pool.totalShares
+    weight: FVal  # Pool.totalWeight
+    # Custom fields
+    user_balance: FVal  # PoolShare.balance
+    asset: Optional[Asset] = None
+
+
+class KnownAsset(NamedTuple):
```

Comment:
As I wrote above this is probably not needed. You can just use `EthereumToken`

Reply:
done